docs-src/concepts.md

```diff
@@ -112,42 +114,123 @@ TODO
 
 ### GatewayClass
 
-`GatewayClass` is cluster-scoped resource defined by the infrastructure ops.
-This resource represents a category of Gateways that can be instantiated.
+`GatewayClass` ([source code][gatewayclass-src]) is cluster-scoped resource
+defined by the infrastructure ops. This resource represents a class of Gateways
+that can be instantiated. 
+
+[gatewayclass-src]: https://github.com/kubernetes-sigs/service-apis/blob/master/api/v1alpha1/gatewayclass_types.go
+
+> Note: this serves the same function as the `networking.IngressClass` resource.
 
 ```yaml
-apiVersion: networking.x-k8s.io/v1alpha1
 kind: GatewayClass
 metadata:
-  name: from-internet
-controller:                          # links to a custom resource
-  apiGroup: networking.acme.io       # that implements this class. 
-  kind: cloud-lb
-  name: from-internet
+  name: cluster-gateway
+spec:
+  controller: "acme.io/gateway-controller"
 ```
 
-A sample controller file:
+We expect that one or more `GatewayClasses` will be created by the infrastructure
+provider for the user. It allows decoupling of which mechanism (e.g. controller)
+implements the `Gateways` from the user. For instance, an infrastructure
+provider may create two `GatewayClasses` named `internet`, `private` to reflect
+`Gateways` that define Internet-facing vs private, internal applications.
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: CloudLB
+kind: GatewayClass
+metadata:
+  name: internet
+  ...
+---
+kind: GatewayClass
 metadata:
-  name: from-internet
-networkTier: auto
-ipAllocation: auto
+  name: private
+  ...
 ```
 
-We expect that one or more GatewayClasses will be created by the infrastructure
-provider for the user as part of their cloud infrastructure. If the cluster
-operator wishes to customize their `GatewayClasses`, they are free to do so,
-interacting with the custom resource parameters. Example parameters
-`GatewayClass.controller` resource will include cluster-level defaults and
-constraints.
+The user of the classes will not need to know *how* `internet` and `private` are
+implemented, whether they are cluster-based or cloud-based LBs.
 
 > The community may converge around a common set of GatewayClass names with
 > well-known meanings (e.g. `internet`, `private`) to ease interoperability
 > between platforms.
 
+#### GatewayClass parameters
```

Comment:
Is there another use-case/example to better understand when to use different classes?
`internet` and `private` are very similar and I can't understand the part of how community will converge to some common names for GatewayClasses.

I'm not too familiar with Storage in k8s but the storage classes there are almost vendor specific and that is very to understand:
https://kubernetes.io/docs/concepts/storage/storage-classes/


Also, let's assume we do converge on names, how can multiple Ingress Controller providing the `internet` GatewayClass co-exist now? I don't see a way that would be possible.

Reply:
These names are meant to be maintained by the cluster infrastructure provider to maintain consistency across their infrastructure.

Providers are free to ship with canonical GatewayClass'es for their products just like how the controllers look at the `Ingress.spec.class` field today to figure out who is in charge of managing the Gateway (and associated objects). e.g. to replicate the current Ingress setup, you would have the following GatewayClass resources  `nginx`, `gce`, `gce-internal`, `kong`, etc. w/ the appropriate parameters.

Why the admin will rename the objects comes when the infrastructure provider wants to ship a full stack (e.g. kubernetes as a platform) along with support for Gateway APIs, but also views how Gateways are implemented as an internal detail, e.g. today they use nginx, tomorrow F5 LB appliance, etc. In this instance, the users just care about the class of LB, e.g. it is to expose to private network internally, on the Internet etc.

Reply:
@bowei Did you thought about how to replace a given gateway class at runtime?
I guess a "simple" change will cause downtime, because 1 GW is already removing the routes while the other is not yet ready and DNS is in the middle switched maybe and depending on http clients and DNS caching it will switch later.
Maybe the field should not be allowed to change.

Reply:
Right now, this will require the admin to delete an recreate the GatewayClass. It's hard to conceive of infra that will be able to bridge that action transparently and feels for certain hard to make that a requirement for implementers.